### PR TITLE
Closes #130 Shared API Gateway and maxAge property support for HTTP Endpoints

### LINKED ISSUE #130: Shared API Gateway support for HTTP Endpoints
# This is a Feature Proposal

Please add support for [Shared API Gateway](https://serverless.com/framework/docs/providers/aws/events/apigateway/#share-api-gateway-and-api-resources)

## Description

* Add shared api gateway support
* No config required

## Additional Data
* ***Serverless Framework Core Version you're using***: 1.28.0
* ***The Plugin Version you're using***: latest
* ***Operating System***: Mac OS X

### PR DESCRIPTION
**Pull request by vlewin**
Closes #130 Shared API Gateway support for HTTP Endpoints

### CODE REVIEW

**Comment by vlewin** [comment]
@horike37 All tests are green but coveralls seems to be unhappy

**Comment by vlewin** [comment]
@horike37 All green!

**Comment by horike37** [comment]
@vlewin 
Great! Can I start to review this PR?

File: lib/deploy/events/apiGateway/cors.js [hunk 1]
```diff
@@ -24,6 +25,16 @@ module.exports = {
         'Access-Control-Allow-Credentials': `'${config.allowCredentials}'`,
       };
 
+      // Enable CORS Max Age usage if set
+      if (_.has(config, 'maxAge')) {
```
Comment (horike37) [inline lib/deploy/events/apiGateway/cors.js]: Seems that maxAge option doesn't work.

Comment (vlewin) [reply lib/deploy/events/apiGateway/cors.js]: @horike37 Good catch! maxAge property issue is fixed.

Comment (vlewin) [reply lib/deploy/events/apiGateway/cors.js]: cloudformation-template-update-stack.json

bildschirmfoto 2018-07-25 um 21 05 09

API Gateway Integration Response

bildschirmfoto 2018-07-25 um 21 06 37

**Comment by horike37** [comment]
@vlewin 
Thank you for the fixing!
That looks perfect!

**Comment by vlewin** [comment]
@horike37 Thank you for accepting and awesome plugin!

**Comment by horike37** [comment]
Great work @vlewin You are Rock!!!

**Pull request merged**